docs/setup-guide/build_k8s_from_source/README.md

```diff
@@ -0,0 +1,145 @@
+{
+  "exec-opts": ["native.cgroupdriver=systemd"],
+  "log-driver": "json-file",
+  "log-opts": {
+    "max-size": "100m"
+  },
+  "storage-driver": "overlay2"
+}
+EOF
+```
+- Run `sudo systemctl daemon-reload`
+- Run `sudo systemctl restart docker`
+
+## Install GoLang
+- Run `cd ~`
+- Run `sudo apt-get -y install make gcc jq`
+- Run `wget https://redirector.gvt1.com/edgedl/go/go1.17.1.linux-amd64.tar.gz -O go1.17.1.tar.gz`
+- Run `tar -xzvf go1.17.1.tar.gz`
+- Run `mkdir gopath`
+- Append the following lines to `~/.bashrc`
+  ```
+  export GOROOT=$HOME/go
+  export GOPATH=$HOME/gopath
+  export PATH=$GOROOT/bin:$PATH
+  ```
+- Run `source ~/.bashrc`
+- Check `go version`
+
+## Build Kubernetes from source
+- Run `sudo apt -y install conntrack socat`
+- Run `mkdir -p $GOPATH/src/k8s.io`
+- Run `cd $GOPATH/src/k8s.io`
+- Run `git clone https://github.com/kubernetes/kubernetes.git`
+- Run `cd kubernetes`
+- Run `git fetch --all --tags`
+- Run `git checkout tags/v1.21.0`
+- Run `make clean`
+- Run `make all`
+- Install CNI plugins by running:
+```
+CNI_VERSION="v0.8.2"
+ARCH="amd64"
+sudo mkdir -p /opt/cni/bin
+curl -L "https://github.com/containernetworking/plugins/releases/download/${CNI_VERSION}/cni-plugins-linux-${ARCH}-${CNI_VERSION}.tgz" | sudo tar -C /opt/cni/bin -xz
+```
+- Install crictl by running:
+```
+DOWNLOAD_DIR=/usr/local/bin
+sudo mkdir -p $DOWNLOAD_DIR
+CRICTL_VERSION="v1.22.0"
+ARCH="amd64"
+curl -L "https://github.com/kubernetes-sigs/cri-tools/releases/download/${CRICTL_VERSION}/crictl-${CRICTL_VERSION}-linux-${ARCH}.tar.gz" | sudo tar -C $DOWNLOAD_DIR -xz
+```
+- Run `sudo ln -s $GOPATH/src/k8s.io/kubernetes/_output/bin/kubelet /usr/bin/`
+- Run `sudo ln -s $GOPATH/src/k8s.io/kubernetes/_output/bin/kubeadm /usr/bin/`
+- Run `sudo ln -s $GOPATH/src/k8s.io/kubernetes/_output/bin/kubectl /usr/bin/`
+- Run `sudo ln -s $GOPATH/src/k8s.io/kubernetes/_output/bin/kubelet /usr/local/bin/`
+- Run `sudo ln -s $GOPATH/src/k8s.io/kubernetes/_output/bin/kubeadm /usr/local/bin/`
+- Run `sudo ln -s $GOPATH/src/k8s.io/kubernetes/_output/bin/kubectl /usr/local/bin/`
+- Add a `kubelet` systemd service by running
+```
+RELEASE_VERSION="v0.4.0"
+curl -sSL "https://raw.githubusercontent.com/kubernetes/release/${RELEASE_VERSION}/cmd/kubepkg/templates/latest/deb/kubelet/lib/systemd/system/kubelet.service" | sed "s:/usr/bin:${DOWNLOAD_DIR}:g" | sudo tee /etc/systemd/system/kubelet.service
+sudo mkdir -p /etc/systemd/system/kubelet.service.d
+curl -sSL "https://raw.githubusercontent.com/kubernetes/release/${RELEASE_VERSION}/cmd/kubepkg/templates/latest/deb/kubeadm/10-kubeadm.conf" | sed "s:/usr/bin:${DOWNLOAD_DIR}:g" | sudo tee /etc/systemd/system/kubelet.service.d/10-kubeadm.conf
+```
+- Add `Environment="KUBELET_EXTRA_ARGS=--fail-swap-on=false"` in `/etc/systemd/system/kubelet.service.d/10-kubeadm.conf`
+- Enable and start by running `sudo systemctl enable --now kubelet`
+- Run
+```
+IPADDR="<your_machine_ip>"
```

Comment:
hostname -I | cut -d' ' -f1 is an easy way to get the local ip

Reply:
changed to IPADDR=$(hostname -I | cut -d ' ' -f1)